### PR TITLE
Updated interfaces

### LINKED ISSUE #360: Add interface for space contract

### PR DESCRIPTION
**Pull request by Orland0x**
Added a interface for the space and space factory contracts. Removed the `execute` method from the Authenticator interface as its not an external function.

closes #360

### CODE REVIEW

File: contracts/starknet/Interfaces/IAuthenticator.cairo [hunk 1]
```diff
@@ -1,7 +1,6 @@
 %lang starknet
 
+// Authenticator contracts have no standard interface, we leave this here for completeness
 @contract_interface
 namespace IAuthenticator {
-    func execute(target: felt, function_selector: felt, calldata_len: felt, calldata: felt*) {
-    }
```
Comment (pscott) [inline contracts/starknet/Interfaces/IAuthenticator.cairo]: I'm not sure why we have this? What's the point of an Interface if there's no function in it?

File: contracts/starknet/Interfaces/ISpaceAccount.cairo [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    func getPublicKey() -> (publicKey: felt) {
+    }
+
+    func supportsInterface(interfaceId: felt) -> (success: felt) {
+    }
+
+    func setPublicKey(newPublicKey: felt) {
+    }
+
+    func isValidSignature(hash: felt, signature_len: felt, signature: felt*) -> (isValid: felt) {
```
Comment (pscott) [inline contracts/starknet/Interfaces/ISpaceAccount.cairo:19]: Really sucks to be using `camelCase` and then `snake_case` for `signature_len` and `signature` but I guess there's no way around it. Feels terribly wrong tho hahaha

**Review by pscott — APPROVED** [review]

File: contracts/starknet/Interfaces/IAuthenticator.cairo (hunk 1 above)
Comment (Orland0x) [reply contracts/starknet/Interfaces/IAuthenticator.cairo]: Okay ill remove

File: contracts/starknet/Interfaces/ISpaceAccount.cairo (hunk 2 above)
Comment (Orland0x) [reply contracts/starknet/Interfaces/ISpaceAccount.cairo:19]: agreed

**Pull request merged**